vulnerable.cpp

```
@@ -1 +1,108 @@
+            char sensitiveData[50] = {0};
+            sensitiveFile.getline(sensitiveData, 50);
+            cout << "Sensitive Data: " << sensitiveData << endl;
+        }
+
+        // 輸出硬編碼的認證資訊（資訊洩漏風險）
+        cout << "Admin Credentials: " << adminUser << " / " << adminPass << endl;
+
+        // 模擬協定解析時，未檢查來源資料長度就進行複製操作
+        char packet[8] = "12345678";
+        char command[5] = {0};
+        strncpy(command, packet, 10);  // 複製量超出 command 陣列大小
```

Cppcheck (2.10-2)

[error] Buffer is accessed out of bounds

(bufferAccessOutOfBounds)